packages/block-library/src/embed/transforms.native.js

```diff
@@ -1,9 +1,11 @@
 /**
  * Internal dependencies
  */
+import webTransforms from './transforms.js';
 import transformationCategories from '../transformationCategories';
 
 const transforms = {
+	...webTransforms,
```

Comment:
re-enabling the webTransforms allows the pasted URL to be transformed to an Embed block. The bug that was logged here https://github.com/wordpress-mobile/gutenberg-mobile/issues/3762 no longer occurs because there are actions that allow the user to determine how the URL should be presented.